development/cpp/index.rst

```diff
@@ -18,3 +18,4 @@ Engine development
    custom_resource_format_loaders
    custom_audiostreams
    custom_godot_servers
+   vulkan/index
```

Comment:
I know we're going back and forth but I'm not sure if we really need a dedicated `vulkan` folder/separate table of content.

Do we intend to have more documentation page relative to Vulkan development? Otherwise this could be `development/cpp/vulkan_validation_layers.rst` directly, without adding an extra level.

If we plan to write more documentation about Vulkan, or about rendering, then making a dedicated Vulkan (or Rendering) sub-section is fine with me. CC @godotengine/rendering

It's not necessarily blocking for this PR though, we can always merge and move the file around later if we change our mind (that just means a 404 on the original link unless we set up a redirection).

Reply:
@akien-mga I think having a dedicated vulkan folder makes sense.
For one setting up vulkan on platform like macos (e.g: moltenVk) is non-trivial so some additional setup documentation could be added in this section.

In addition, given that the standard is still evolving and there might be some large difference between desktop and mobile, this section could also be used to highlight that.

Reply:
I was under the impression that we all agreed on Fredia's suggestion, that Vulkan is a big topic and it deserves a dedicated category. I think that there will be more docs regarding Vulkan in the future.

Reply:
Sounds good, let's go with this then :)